src/main/java/io/nats/streaming/StreamingConnectionImpl.java

```diff
@@ -552,14 +552,25 @@ protected void processAck(Message msg) {
     TimerTask createAckTimerTask(String guid) {
         TimerTask task = new java.util.TimerTask() {
             public void run() {
-                processAckTimeout(guid);
+                try {
+                    processAckTimeout(guid);
+                } catch (Exception e) {
+                    // catch exception to prevent the timer to be closed
+                    logger.error("stan: error encountered during processAckTimeout, will cancel this timer task", e);
+                    // cancel this task
+                    cancel();
+                }
             }
         };
         return task;
     }
 
     protected void processAckTimeout(String guid) {
         AckClosure ackClosure = removeAck(guid);
+        if (ackClosure == null) {
```

Comment:
Indeed, the code should check if `ackClosure` is `null` and if so, simply return. It means that the connection has received the publisher ack from the server and started to process it (removed from the map) and has or will invoke the ack handler callback or notify the synchronous publish call.

We have the same problem in Go (I am going to create an issue/fix for that), except that in Go there won't be an issue with the ack closure being null because the publish call use a function closure for the timer and the ack structure is used there. It is wrong though, since we should get the ack from the `removeAck()` call that ensures that only one thread/go routine gets to the pub ack.